web/server.go

```diff
@@ -143,31 +144,60 @@ func checkExists(filepath string) error {
 	return nil
 }
 
-func listenAndServe(noAdmin bool, appsHandler echo.HandlerFunc) error {
-	main, err := CreateSubdomainProxy(echo.New(), appsHandler)
+func listenAndServe(appsHandler echo.HandlerFunc) (*Servers, error) {
+	major, err := CreateSubdomainProxy(echo.New(), appsHandler)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if err = LoadSupportedLocales(); err != nil {
-		return err
+		return nil, err
 	}
 
 	if config.IsDevRelease() {
-		main.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
+		major.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
 			Format: "time=${time_rfc3339}\tstatus=${status}\tmethod=${method}\thost=${host}\turi=${uri}\tbytes_out=${bytes_out}\n",
 		}))
 	}
 
-	errs := make(chan error)
-
-	if !noAdmin {
-		admin := echo.New()
-		if err = SetupAdminRoutes(admin); err != nil {
-			return err
-		}
-		go func() { errs <- admin.Start(config.AdminServerAddr()) }()
+	admin := echo.New()
+	if err = SetupAdminRoutes(admin); err != nil {
+		return nil, err
 	}
 
-	go func() { errs <- main.Start(config.ServerAddr()) }()
-	return <-errs
+	return &Servers{
+		major: major,
+		admin: admin,
+	}, nil
+}
+
+// Servers contains the started HTTP servers and implement the Shutdowner
+// interface.
+type Servers struct {
+	major *echo.Echo
+	admin *echo.Echo
+	errs  chan error
+}
+
+// Start starts the servers.
+func (e *Servers) Start() {
```

Comment:
I am not sure why server .Start method errors have to be handled in go routine ?